packages/devtools_app/lib/src/screens/memory/memory_controller.dart

```diff
@@ -225,6 +230,35 @@ class AllocationSamples {
   }
 }
 
+mixin MemoryControllerMixin<T extends StatefulWidget> on State<T> {
+  bool _isInitialized = false;
```

Comment:
let's get rid of the initialized pattern. I'd have to dig up the PR, but there was one a while back where we discussed replacing the late / initialized pattern with a non-nullable getter instead.
```
  MemoryController get controller => _controller!;
  MemoryController? _controller;

  @override
  void didChangeDependencies() {
    super.didChangeDependencies();

    final newController = Provider.of<MemoryController>(context);
    if (newController == _controller) return;
    _controller = newController;
  }
```

I do like using a Mixin for this though. This will remove several lines of boilerplate from many classes. We should do this for other screens as well.